admin/deployments.go

```diff
@@ -170,6 +170,46 @@ func (s *Service) updateDeployment(ctx context.Context, depl *database.Deploymen
 	return nil
 }
 
+// HIBERNATE free deployments
+func (s *Service) HibernateDeployments(ctx context.Context) error {
+	// Not checking any permissions as its use only internally by scheduled job, will add if required
+	projects, err := s.DB.FindAllProjects(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, proj := range projects {
+		depl, err := s.DB.FindDeployment(ctx, *proj.ProdDeploymentID)
+		if err != nil {
+			fmt.Printf("depl not found for proj %s", proj.Name)
+			continue
+		}
+
+		if depl.Status == database.DeploymentStatusReconciling && time.Since(depl.UpdatedOn) < 30*time.Minute {
+			fmt.Printf("skipping because it is already running\n")
+			continue
+		}
+
+		if proj.ProductionTTL < time.Since(depl.UpdatedOn) {
+			err := s.teardownDeployment(ctx, proj, depl)
```

Comment:
`teardownDeployment` completely deletes the deployment, so we can't update its status after – not sure what the best path forward is here